libeos-update-server/repo.c

```diff
@@ -162,62 +162,104 @@ eos_updater_repo_server_get_property (GObject *object,
 }
 
 static void
-eos_updater_repo_server_set_property (GObject *object,
-                                      guint property_id,
-                                      const GValue *value,
-                                      GParamSpec *spec)
+eus_repo_set_property (GObject      *object,
+                       guint         property_id,
+                       const GValue *value,
+                       GParamSpec   *spec)
 {
-  EosUpdaterRepoServer *server = EOS_UPDATER_REPO_SERVER (object);
+  EusRepo *self = EUS_REPO (object);
 
-  switch (property_id)
+  switch ((EusRepoProperty) property_id)
     {
     case PROP_REPO:
-      g_set_object (&server->repo, g_value_get_object (value));
+      g_set_object (&self->repo, g_value_get_object (value));
       break;
 
+    case PROP_ROOT_PATH:
+      g_clear_pointer (&self->root_path, g_free);
+      self->root_path = g_value_dup_string (value);
+
+      g_assert (self->root_path != NULL);
+
+      /* Add a missing leading slash if the root path is non-empty. */
+      if (self->root_path[0] != '\0' && self->root_path[0] != '/')
+        {
+          g_autofree gchar *tmp = g_strconcat ("/", self->root_path, NULL);
+          g_free (self->root_path);
+          self->root_path = g_steal_pointer (&tmp);
+        }
+
+      /* Drop any trailing slash. */
+      if (strlen (self->root_path) > 1 &&
```

Comment:
Hopefully the compiler is smart enough to optmize all those calls to strlen to a single one.